pandas/tests/reshape/concat/test_concat.py

```diff
@@ -599,3 +599,8 @@ def test_concat_repeated_keys(keys, integrity):
     tuples = list(zip(keys, ["a", "b", "c"]))
     expected = Series([1, 2, 3], index=MultiIndex.from_tuples(tuples))
     tm.assert_series_equal(result, expected)
+
+
+def test_concat_no_items_raises():
+    with pytest.raises(ValueError, match="No objects to concatenate"):
```

Comment:
is there a reference somewhere of discussion on why this should raise rather than return an empty Series/DataFrame?

Reply:
Came up in #40820

Reply:
we don't have any other tests for this?

Reply:
```   def test_concat_exclude_none(self):
        df = DataFrame(np.random.randn(10, 4))

        pieces = [df[:5], None, None, df[5:]]
        result = concat(pieces)
        tm.assert_frame_equal(result, df)
        with pytest.raises(ValueError, match="All objects passed were None"):
            concat([None, None])
```

put near this one

Reply:
> 
> 
> we don't have any other tests for this?

By grepping I did not find that ``concat`` is anywhere invoked on an empty list, besides this new test.

```
$ grep -oPrn 'concat\(\[\]\)' pandas/tests
pandas/tests/reshape/concat/test_concat.py:606:concat([])
```

Reply:
can you put it next to the test as i indicated above